ingest/src/sources/mongodb-university/MongoDbUniversityDataApiClient.ts

```diff
@@ -0,0 +1,179 @@
+import fetch from "node-fetch";
+/**
+  Item in the Thought Industries catalog as returned from the
+  MongoDB University Data API.
+ */
+export interface TiCatalogItem {
+  id: string;
+  ti_id: string;
+  /**
+        IDs for videos associated with the catalog item.
+       */
+  associated_videos: string[];
+  /**
+        Learning format for the catalog item.
+        @example "Unit" || "Course" || "Lab" || "...etc"
+       */
+  learning_format: string;
+  name: string;
+  slug: string;
+  status: string;
+  tags: string[];
+  legacy: boolean;
+  /**
+        Thought Industries microsites associated with the catalog item.
+        @example ["University", "Technical Services"]
+       */
+  microsites: string[];
+  /**
+        IDs of labs associated with the catalog item.
+       */
+  associated_labs: string[];
+  // TODO: is this always null?
+  associated_content: null | any;
+  /**
+        Whether or not the catalog item is in development.
+       */
+  in_development: boolean;
+  created_at: string;
+  /**
+        Sections in the catalog item. This contains metadata
+        about the sections and lessons.
+       */
+  sections: TiCatalogSection[];
+}
+
+interface TiCatalogSection {
+  slug: string;
+  title: string;
+  lessons: TiCatalogLesson[];
+}
+
+interface TiCatalogLesson {
+  slug: string;
+  title: string;
+  labs: string[]; // Assuming 'labs' is an array of strings, similar to 'videos'.
+  videos: string[];
+}
+
+/**
+  MongoDB University Video as returned from the MongoDB University Data API.
+ */
+export interface UniversityVideo {
+  id: string;
+  hashed_id: string;
+  created: string;
+  description: string;
+  duration: number;
+  name: string;
+  progress: number;
```

Comment:
Curious what `progress` means here. Is this an instance of a video for a specific user?